src/lib/machine.ts

```diff
@@ -302,7 +305,9 @@ export class Machine {
     this.client = client
   }
 
-  async listMachines(app_name: ListMachineRequest): Promise<MachineResponse[]> {
+  async listMachines(
+    app_name: ListMachineRequest
+  ): Promise<APIResponse<MachineResponse[]>> {
```

Comment:
What's the reason for wrapping errors in the response object instead of throwing? This would be a breaking change for existing users of this library which I'd like to avoid.

A few other options
1. create another wrapper class
2. bump the major version